src/components/transport_manager/src/tcp/tcp_client_listener.cc

```diff
@@ -289,6 +289,7 @@ void TcpClientListener::Loop() {
       strncpy(device_name,
               inet_ntoa(client_address.sin_addr),
               sizeof(device_name) / sizeof(device_name[0]));
+      device_name[sizeof(device_name)-1] = '\0';
```

Comment:
`sizeof` returns the size of the value in bytes. You would need to use the same trick as the above line to get the length (`sizeof(device_name) / sizeof(device_name[0])`). You could likely just set this to a variable and use it in both places.